packages/babel-generator/src/buffer.js

```diff
@@ -14,6 +14,7 @@ export default class Buffer {
     this._indent = format.indent.base;
     this.format = format;
     this.buf = "";
+    this.last = "";
```

Comment:
Very interesting observation. So in your example, is the flatten happens only once, when you access an indexed symbol the first time? Because the time is `0` when you access it second time, when do `buf[buf.length - 1];`?